aspnetcore/blazor/components/render-modes.md

```diff
@@ -0,0 +1,378 @@
+
+Component builder extensions:
+
+* <xref:Microsoft.Extensions.DependencyInjection.RazorComponentsBuilderExtensions.AddServerComponents%2A> adds services to support rendering interactive server components.
+* `AddWebAssemblyComponents` adds services to support rendering interactive WebAssembly components.
+
+<!-- UPDATE 8.0 HOLD
+     <xref:Microsoft.Extensions.DependencyInjection.WebAssemblyRazorComponentsBuilderExtensions.AddWebAssemblyComponents%2A>
+-->
+
+<xref:Microsoft.AspNetCore.Builder.RazorComponentsEndpointRouteBuilderExtensions.MapRazorComponents%2A> discovers available components and specifies the root component for the app, which by default is the `App` component (`App.razor`).
+
+Endpoint convention builder extensions:
+
+* <xref:Microsoft.AspNetCore.Builder.RazorComponentEndpointConventionBuilder.AddServerRenderMode%2A> configures the Server render mode for the app.
+* `AddWebAssemblyRenderMode` configures the WebAssembly render mode for the app.
+
+<!-- UPDATE 8.0 HOLD
+    <xref:Microsoft.AspNetCore.Builder.WebAssemblyRazorComponentsEndpointConventionBuilderExtensions.AddWebAssemblyRenderMode%2A>
+-->
+
+> [!NOTE]
+> For orientation on the placement of the API in the following examples, inspect the `Program` file of an app generated from the Blazor Web App project template. For guidance on how to create a Blazor Web App, see <xref:blazor/tooling>.
+
+Example 1: The following `Program` file API adds services and a render mode for Razor components with only server-side rendering support:
+
+```csharp
+builder.Services.AddRazorComponents()
+    .AddServerComponents();
+```
+
+```csharp
+app.MapRazorComponents<App>()
+    .AddServerRenderMode();
+```
+
+Example 2: The following `Program` file API adds services and a render mode for Razor components with only client-side rendering support:
+
+```csharp
+builder.Services.AddRazorComponents()
+    .AddWebAssemblyComponents();
+```
+
+```csharp
+app.MapRazorComponents<App>()
+    .AddWebAssemblyRenderMode();
+```
+
+Example 3: The following `Program` file API adds services and render modes for Razor components with both server-side and client-side rendering support:
+
+```csharp
+builder.Services.AddRazorComponents()
+    .AddServerComponents()
+    .AddWebAssemblyComponents();
+```
+
+```csharp
+app.MapRazorComponents<App>()
+    .AddServerRenderMode()
+    .AddWebAssemblyRenderMode();
+```
+
+## Apply a render mode to a component instance
+
+To apply a render mode to a component instance use the [`@rendermode` Razor directive attribute](xref:mvc/views/razor#attribute) where the component is used.
+
+> [!NOTE]
+> Component authors should typically design components to support any render mode or hosting model. Components should avoid assumptions on where they are running (server or client) and should degrade gracefully when rendered statically.
+
```

Comment:
I realized this note makes less sense here where we're talking about using components, not authoring them.

```suggestion
```